core/modules/css/KvInterfaceImplMySql.cc

```diff
@@ -0,0 +1,319 @@
+    } else if (errObj.isSet()) {
+        throw CssError(errObj);
+    }
+
+    transaction.commit();
+}
+
+
+bool
+KvInterfaceImplMySql::exists(std::string const& key) {
+    KvTransaction transaction(_conn);
+
+    std::string val;
+    sql::SqlErrorObject errObj;
+    return _getValFromServer(key, val, errObj);
+}
+
+
+std::vector<std::string>
+KvInterfaceImplMySql::getChildren(std::string const& key) {
+    KvTransaction transaction(_conn);
+
+    // per the documentation: verify the parent key exists
+    if (not exists(key)) {
+        throw NoSuchKey(key);
+    }
+
+    boost::format fmQuery("SELECT kvKey FROM cssData WHERE parentKey='%1%';");
+    fmQuery % key;
+    std::string query = fmQuery.str();
+
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    LOGF(_logger, LOG_LVL_DEBUG, "getChildren - executing query: %1%" % query);
+    if (not _conn.runQuery(query, results, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "getChildren - %1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError((boost::format("getChildren - error:%1% from query:%2%") % errObj.errMsg() % query).str());
+    }
+
+    errObj.reset();
+    std::vector<std::string> ret;
+    if (not results.extractFirstColumn(ret, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "getChildren - failed to extract children from %1% failed with err:%2%" % query % errObj.errMsg());
+        throw CssError((boost::format("getChildren - failed to extract children error:%1% from query:%2%") % errObj.errMsg() % query).str());
+    }
+
+    transaction.commit();
+
+    return ret;
+}
+
+
+void
+KvInterfaceImplMySql::deleteKey(std::string const& key) {
+    KvTransaction transaction(_conn);
+
+    boost::format fmQuery("DELETE FROM cssData WHERE kvKey='%1%';");
+    fmQuery % key;
+    std::string query = fmQuery.str();
+
+    // run query
+    sql::SqlErrorObject errObj;
+    sql::SqlResults resultsObj;
+    LOGF(_logger, LOG_LVL_DEBUG, "Executing query: %1%" % query);
+    if (not _conn.runQuery(query, resultsObj, errObj)) {
+        LOGF(_logger, LOG_LVL_ERROR, "SQL DELETE FROM failed: %1%" % query);
+        throw CssError(errObj);
+    }
+
+    // limit the row count to 1 - there should not be multiple rows for any key.
+    auto affectedRows = resultsObj.getAffectedRows();
+    if (affectedRows < 1) {
+        LOGF(_logger, LOG_LVL_ERROR, "SQL DELETE FROM failed (no such key): %1%" % query);
+        throw NoSuchKey(errObj);
+    }
+    else if (affectedRows > 1) {
+        LOGF(_logger, LOG_LVL_ERROR, "SQL DELETE FROM failed (too many (%1%) rows deleted): %2%" % affectedRows % query);
+        throw CssError("deleteKey - unexpectedly deleted more than 1 row.");
+    }
+
+    transaction.commit();
+}
+
+
+std::string
+KvInterfaceImplMySql::_get(std::string const& key, std::string const& defaultValue, bool throwIfKeyNotFound) {
+    std::string val;
+    sql::SqlErrorObject errObj;
+    if (not _getValFromServer(key, val, errObj)) {
```

Comment:
Same here, false returned from _getValFromServer means two things, you want to check `errObj` and always throw (something that is not NoSuchKey) if error is set in it.